generators/FairBaseMCGenerator.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * FairBaseMCGenerator.h
+ *
+ *  Created on: 16 mar 2020
+ *      Author: Daniel Wielanek
+ *		E-mail: daniel.wielanek@gmail.com
+ *		Warsaw University of Technology, Faculty of Physics
+ */
+#ifndef FAIR_BASEMCGENERATOR_H_
+#define FAIR_BASEMCGENERATOR_H_
+
+#include "FairGenerator.h"
+
+/**
+ * abstract class for generators that use "macro input" to generate partcles
+ */
+class FairBaseMCGenerator : public FairGenerator{
+public:
+	enum eVertexSmear{
+		kBox,
+		kGauss,
+		kExp
+	};
+	FairBaseMCGenerator();
```

Comment:
I don't see why we need the copy constructor as well as the assignment operator. I don't see the need to copy an existing operator. In my opinion if more than one generator is needed one should create each one separately. If these constructors are really needed please change is to   

FairBaseMCGenerator(const FairBaseMCGenerator) = default;
FairBaseMCGenerator& operator=(const FairBaseMCGenerator) =default;

and delete the implementation. Since all data members are simple types there is no need to implement the constructors explicitly.

Reply:
Hi Florian,
I think not it should be fine. I also remove change constructors for FairBox, this generator also contains only simple types.